android/build.gradle

```diff
@@ -64,9 +64,6 @@ dependencies {
       implementation "com.google.firebase:firebase-iid"
   } else if(firebaseIidVersion){
       implementation "com.google.firebase:firebase-iid:${firebaseIidVersion}"
-  }else{
-      def iidVersion = safeExtGet('googlePlayServicesIidVersion', safeExtGet('googlePlayServicesVersion', '17.0.0'))
-      implementation "com.google.android.gms:play-services-iid:$iidVersion"
```

Comment:
You have made this change so that it is not even possible to make this backwards compatible for people by removing this logic completely

The more developer-friendly way to do this is not to just delete everything and YOLO a release because it does not affect yout

The more developer-friendly way is to at least make it so developers can maintain the old functionality if they need it

Move the safeExtGet of `googlePlayServicesIidVersion` out of the if/else if/else area, put it up with the other two, and have it also default to null if it does not exist

Then change the final `else` to an else if (googlePlayServicesIidVersion) and pull in the dependency if it was non-null

Leave the code below that allows fetching GMS version if people did set a googlePlayServicesIidVersion

Now in the changelog we can issue a breaking change *but* we can say:

"Np longer add GMS Instance ID fetch as default fallback. If you relied on this functionality, update your gradle build file to define `googlePlayServicesIidVersion` if you have not already, previously it defaulted to `17.0.0`"

Which is much better than "If you relied on this functionality, you can't update this module ever"

Reply:
Hi, @mikehardy ,
Thanks for the detailed feedback!
I've made all the suggested changes and updated the PR accordingly.